### PR TITLE
Refresh and add examples for tidyr::fill() 

### LINKED ISSUE #858: Refresh `fill()` docs
* [ ] Proofread title, description, and parameters 
* [ ] Proofread comments in examples 
* [ ] Ensure comment headings used to divide into scannable sections 
* [ ] Flag examples that seem overly complex for further review 
* [ ] Switch out mtcars/iris for something more interesting, or a smaller dataset constructed specifically to illustrate one point. 
* [ ] Review size of example outputs to ensure that you can easily see important rows and cols. 
* [ ] Datasets should always be tibbles to avoid use of `as_tibble()` distracting from the main point

### PR DESCRIPTION
**Pull request by jthomasmock**
Fixes #858

### CODE REVIEW

File: R/fill.R [hunk 1]
```diff
@@ -19,8 +19,94 @@ NULL
 #'   or "updown" (first up and then down).
 #' @export
 #' @examples
-#' df <- data.frame(Month = 1:12, Year = c(2000, rep(NA, 11)))
-#' df %>% fill(Year)
+#' # Here we have some spreadsheet data where the analyst has only updated year
+#' # when the year has changed, leading to a consistent placement of year at the top
+#' # followed by trailing NAs up to the next year
+#'
+#' sales <- tibble(quarter = rep(c("Q1", "Q2", "Q3", "Q4"), 4),
```
Comment (hadley) [inline R/fill.R]: Can you please indent long lines using our standard rules? https://style.tidyverse.org/syntax.html#long-lines

You might also consider using a `tribble()` to make the placement of the missing values more clear.

File: R/fill.R [hunk 2]
```diff
@@ -19,8 +19,94 @@ NULL
...
+#'                          2002, rep(NA, 3),
+#'                          2004, rep(NA, 3)
+#'                 ),
+#'                 sales = c(66013, 69182, 53175, 21001, 46036, 58842, 44568, 50197,
+#'                           39113, 41668, 30144, 52897, 32129, 67686, 31768, 49094)
+#' )
+#'
+#' sales
+#'
+#' # We can call `fill()` and only need to specify the column to fill since the
```
Comment (hadley) [inline R/fill.R]: I think overall you've provided a bit too much explanation (which would be suitable for a vignette, but not docs) — can you please make it a bit more concise?

File: R/fill.R (hunk 1 above)
Comment (jthomasmock) [reply R/fill.R]: Will do, thanks for feedback!

File: R/fill.R [hunk 3]
```diff
@@ -19,8 +19,91 @@ NULL
...
+#'   "Q2",    NA,  41668,
+#'   "Q3",    NA,  30144,
+#'   "Q4",    NA,  52897,
+#'   "Q1",  2004,  32129,
+#'   "Q2",    NA,  67686,
+#'   "Q3",    NA,  31768,
+#'   "Q4",    NA,  49094
+#' )
+#'
+#' # `fill()` will default to replacing missing data from top to bottom
```
Comment (hadley) [inline R/fill.R]:
```suggestion
#' # `fill()` defaults to replacing missing data from top to bottom
```

File: R/fill.R [hunk 4]
```diff
@@ -19,8 +19,91 @@ NULL
...
+#' sales %>% fill(year)
+#'
+#'
```
Comment (hadley) [inline R/fill.R]: Can you please use one empty line between examples?

File: R/fill.R [hunk 5]
```diff
@@ -19,8 +19,91 @@ NULL
...
+#' # Value (pet_type) is missing above
+#' tidy_pets <- tibble::tribble(
+#'   ~rank, ~pet_type,                          ~breed,
```
Comment (hadley) [inline R/fill.R]: `~breed` needs to come to the left a little. And I think maybe you could reduce it to 6 breeds per type?

File: R/fill.R [hunk 6]
```diff
@@ -19,8 +19,91 @@ NULL
...
+#' )
+#'
+#' # The values are inconsistently missing by position within the group
+#' # Use .direction = "downup" to fill missing values in both directions
+#' squirrels %>%
+#'   dplyr::group_by(group) %>%
+#'   fill(n_squirrels, .direction = "downup") %>%
+#'   dplyr::ungroup()
+#'
+#' # Using `.direction = "updown"` accomplishes the same goal in this example
```
Comment (hadley) [inline R/fill.R]: I think you could omit the code here (but leave the comment)

**Pull request merged**